### ISSUE TITLE
Emacs.app not working from brew install

### ISSUE THREAD

**Issue by latkins** [railwaycat/emacs-mac-port#84]
I installed via brew, as described, but have found that trying to run Emacs.app does nothing, with no error message. Running from the terminal works fine. Does anyone have some idea why this might be?

**Comment by devinrhode2** [comment]
I would try brew doc, follow it's suggestions, `brew uninstall emacs-mac` and then follow the reinstall instructions. Post install you should follow the note to run `brew linkapps emacs-mac`. Then brew doc, see if it works, and if it doesn't, report back with what the brew doc said.

**Comment by yesimon** [comment]
I am also getting this error - I did recently upgrade to El Capitan and the previously installed version emacs-24.5-z-mac-5.11 works, but emacs-24.5-z-mac-5.13 does not. 

Starting emacs via the terminal works fine however. Is there any way to debug an `*.app` that doesn't work / crashes immediately?

Is there a better place to report bugs than here on the mirror?

**Comment by railwaycat** [comment]
Hi @yesimon, did you copy older version `Emacs.app` to `/Applications` or have outdated symbolic link?

**Comment by yesimon** [comment]
Symlink to new doesn't work, but works with the old one. Clicking on new `Emacs.app` doesn't work, but works on the old one. What's interesting is that `brew linkapps` links to the versioned `Emacs.app` in the Cellar, and not to the path in `opt` - which might be a bug as well in homebrew.

**Comment by railwaycat** [comment]
Even clicking on the `Emacs.app` under `opt/emacs-mac/` not working? Interesting...

`brew linkapps` thing maybe some history reason. Homebrew not using `opt` directory from the beginning.

**Comment by yesimon** [comment]
Nope, but clicking on the executable in the app works (which opens up a terminal that runs emacs, so yes, starting emacs from terminal also works).

**Comment by napopa** [comment]
can confirm this on el capitan 10.11.1. I went back to emacs-mac 5.11 and the icon is working again

**Comment by crmne** [comment]
Same here as @napopa

**Comment by joefiorini** [comment]
I'm seeing this issue too. The terminal version works fine, but when I try to launch the GUI it crashes before ever displaying a UI. After a crash, opened up console and I saw this:

> 11/25/15 11:05:57.648 PM com.apple.xpc.launchd[1]: (org.gnu.Emacs.367712[85189]) FairPlay decryption failed on binary.

Not sure how to troubleshoot it from there, couldn't find much helpful searching for that error. @yesimon @napopa @crmne Can you all open Console.app, then try to open emacs from the icon and see if you get the same error?

**Comment by mattly** [comment]
I'm having the same problem with the same console error. Using `open` from the terminal works fine but is far less than ideal.

**Comment by crmne** [comment]
@joefiorini I confirm.

> 26/11/15 18:55:56,402 com.apple.xpc.launchd[1]: (org.gnu.Emacs.1312[29562]) FairPlay decryption failed on binary.

**Comment by napopa** [comment]
@joefiorini confirmed 

`26/11/15 13:04:43,837 com.apple.xpc.launchd[1]: (com.apple.xpc.launchd.oneshot.0x1000000f.Emacs[11129]) FairPlay decryption failed on binary.`

See this related 
issue https://github.com/railwaycat/homebrew-emacsmacport/issues/43

**Comment by yesimon** [comment]
@joefiorini Same error here

**Comment by railwaycat** [comment]
Sorry for the late reply!

But I still not have an idea of the root cause of this issue and I'm not able to reproduce this on both 10.11 and 10.10 system, with or without my Apple ID/Dev key deployed :( 

I turned to Google and looks some other open source software facing this error on OS X recently but nothing more helpful. I would suspect if there is recent change on system side result this? I will pushing 24.5-mac-5.14 from upstream soon and please let me know if this issue still there after upgrade. 

some brain storm:

1. For pre-built, I'm using my Apple Dev ID to sign bundle from 24.5-mac-5.11, now I start to rethink if this is a good idea. Not sure if there's any relation to this issue.

2. I'm going to drop 10.9 support for pre-built...

3. If Homebrew works as expect, installing from Homebrew should exactly the same as build and install the original. I don't think this is an issue from upstream but it's worth to have a try if I could make it reproduce. 

4. Make sure the `Emacs.app` in `/Applications/` matches the active version installed in Homebrew if you copy bundle from `Cellar/emacs-mac/emacs-24.5-z-mac-5.xx/`; mismatched bundle sometime works sometime not....

5. Not sure if related to the recent App Store problem :-/

**Comment by yesimon** [comment]
Just to add another datapoint. I did a clean install of El Capitan to solve unrelated issues. The first install of emacs-mac on basically a clean machine had a working icon. But when I reinstalled it, I got the same problem as before.

**Comment by mattly** [comment]
I can confirm @yesimon's data point – the first install of emacs-mac from homebrew on a fresh system works fine. The second does not.

**Comment by joefiorini** [comment]
fwiw, I just upgraded to 5.15 and that opens for me via both the original location (under `/usr/local/bin`) and the symlink under `/Applications`.

**Comment by joefiorini** [comment]
Since it appears symlinked apps don't show up in Spotlight, I just tried copying the .app file from `/usr/local/Cellar/emacs-mac/...` to `/Applications` and it shows in Spotlight but then I get the FairPlay error again. Based on that, I'm guessing this has something to do with the location of the `.app` file, but not sure what...

**Comment by railwaycat** [comment]
It happened to me for once recently. After tried almost all methods and going to give up, I emptied trash bin and reboot. 

Then it fixed...


I guess it's Finder's wrong cache? Not sure if this works for others. BTW, I usually copy `.app` file from cellar to `/Applications`.
